router/core/graphql_prehandler.go

```diff
@@ -597,6 +602,36 @@ func (h *PreHandler) handleOperation(req *http.Request, buf *bytes.Buffer, httpO
 
 	httpOperation.traceTimings.EndPlanning()
 
+	/**
+	* Validate that the planned query doesn't exceed the maximum query depth configured
+	 */
+
+	if h.maxQueryDepth > 0 {
+		_, maxQueryDepthSpan := h.tracer.Start(req.Context(), "Operation - Checking Query Depth",
```

Comment:
I'm open for this but in my opinion it is not a good idea to create a span for this prematurely. You can add the error to the root span.